### ISSUE TITLE
Error BufImgSurfaceData

### ISSUE THREAD

**Issue by lew** [bisq-network/bisq#445]
Error
sun.awt.image.BufImgSurfaceData cannot be cast to sun.java2d.xr.XRSurfaceData

appears at seemingly random intervals

**Comment by ManfredKarrer** [comment]
Which OS are you using? 
If Linux, which desktop are u using?
Can you send me the logs?
Does that happen after sleep mode?

**Comment by mrosseel** [comment]
Java bug, will be solved in 8u112.

jdk bug:
https://bugs.openjdk.java.net/browse/JDK-7172749

fix:
http://hg.openjdk.java.net/jdk9/client/jdk/rev/2bafa85a456b

Overriding the jdk classes is possible but a bit tricky (classpath juggling), let me know if you want me to try.

**Comment by ManfredKarrer** [comment]
@mrosseel Thanks for the input! 
Do u have any idea or estimation when 8u112 will be released? 
It is not a super critical bug, mostly only appearing after monitor setting change or after standby. So would suggest to wait, if its more effort to get a quick fix in.

**Comment by mrosseel** [comment]
October 2016 should be the GA of 8u112, it's noted in the SortedList implementation which can then also be retired. Waiting would be my suggestion as well, if it doesn't happen too much.

**Comment by ManfredKarrer** [comment]
I just discovered an issue. The sorting of the table does nor work anymore. I use sorted by data in most tables but that is not applied and the sort icons also does not get displayed nor interact by clicking the header. I assume they use a static reference to the original SortedList class. 
Do you know how to ensure compilation of our class if we use the use the original package path?

**Comment by mrosseel** [comment]
Comment should be on other ticket

**Comment by ManfredKarrer** [comment]
upps, yes, i move it...

**Comment by ManfredKarrer** [comment]
Just tested on Ubuntu 14 64bit and get still the exception. It can be reproduced by scaling the VM window a bit.
Thought not absolutely sure if the new class get used, but I can confirm that the new SortedListClass is included.

java.lang.ClassCastException: sun.awt.image.BufImgSurfaceData cannot be cast to sun.java2d.xr.XRSurfaceData
    at sun.java2d.xr.XRPMBlitLoops.cacheToTmpSurface(XRPMBlitLoops.java:145)
    at sun.java2d.xr.XrSwToPMBlit.Blit(XRPMBlitLoops.java:353)
    at sun.java2d.pipe.DrawImage.blitSurfaceData(DrawImage.java:959)
    at sun.java2d.pipe.DrawImage.renderImageCopy(DrawImage.java:577)
    at sun.java2d.pipe.DrawImage.copyImage(DrawImage.java:86)
    at sun.java2d.pipe.DrawImage.copyImage(DrawImage.java:1030)
    at sun.java2d.pipe.ValidatePipe.copyImage(ValidatePipe.java:196)
    at sun.java2d.SunGraphics2D.copyImage(SunGraphics2D.java:3229)
    at sun.java2d.SunGraphics2D.drawImage(SunGraphics2D.java:3270)
    at sun.java2d.SunGraphics2D.drawImage(SunGraphics2D.java:3214)
    at sun.awt.X11.XTrayIconPeer$IconCanvas.paint(XTrayIconPeer.java:601)
    at sun.awt.RepaintArea.paintComponent(RepaintArea.java:264)
    at sun.awt.X11.XRepaintArea.paintComponent(XRepaintArea.java:64)
    at sun.awt.RepaintArea.paint(RepaintArea.java:240)
    at sun.awt.X11.XComponentPeer.handleEvent(XComponentPeer.java:584)
    at java.awt.Component.dispatchEventImpl(Component.java:4967)
    at java.awt.Component.dispatchEvent(Component.java:4713)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
    at java.awt.EventQueue$4.run(EventQueue.java:731)
    at java.awt.EventQueue$4.run(EventQueue.java:729)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

**Comment by KanoczTomas** [comment]
[bitsquare.txt](https://github.com/bitsquare/bitsquare/files/487098/bitsquare.txt)

Hello, I believe I have the same issue ... here is the error from clipboard ...

sun.awt.image.BufImgSurfaceData cannot be cast to sun.java2d.xr.XRSurfaceData

I have attached the log file. I am running in a VM using VirtualBox 5.0.26 at the moment, 

tk@tk-VirtualBox:~/.local/share/Bitsquare$ uname -a
Linux tk-VirtualBox 4.4.0-31-generic #50~14.04.1-Ubuntu SMP Wed Jul 13 01:07:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
tk@tk-VirtualBox:~/.local/share/Bitsquare$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.5 LTS
Release:    14.04
Codename:   trusty

I also frequently loose all P2P peers, this started happening after I published an offer, takes a few minutes to get back to the network again. Btw the same issue arises (either with peers or with that error above) when I run on normal machine, not VM. The release is the same there, both are xubuntus, so using Xfce ... 

hope this helps ...

Edit: oh and Bisquare version is 0.4.9.6
